aioamqp/protocol.py

```diff
@@ -317,9 +321,13 @@ def channel(self, **kwargs):
         """Factory to create a new channel
 
         """
-        if not len(self.channels_ids):
-            raise exceptions.NoChannelAvailable()
-        channel_id = self.channels_ids.pop(0)
+        try:
+            channel_id = self.channels_ids_free.pop()
+        except KeyError:
+            if self.channels_ids_ceil >= amqp_constants.MAX_CHANNELS:
```

Comment:
previusly the channels available were: `list(range(1, amqp_constants.MAX_CHANNELS + 1))`

That is from `1` to `MAX_CHANNELS_IDS` included

In this patch I kept the same behaviour.

So it was already broken?